.github/workflows/test-demo.yaml

```diff
@@ -0,0 +1,38 @@
+name: Test VM demo
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        distro:
+          - "archlinux:latest"
+          - "debian:12"
+          - "debian:unstable"
+          - "ubuntu:24.04"
+          - "ubuntu:24.10"
+          - "ubuntu:devel"
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v4'
+      - uses: DeterminateSystems/nix-installer-action@v12
+
+      - name: Build projects overview
+        run: nix build .#overview
+
+      - name: Run and test VM
+        run: >
+          docker run
+          --privileged
+          --volume ./result/project/Cryptpad/default.nix:/default.nix
```

Comment:
I guess the next step after this would be to extend this to all demos from the overview.

Reply:
The demo might work for a project but fail for another, though, right? I thought the goal was to make sure that all demos work.

Reply:
Yes, but this shouldn't stop us as the plan is to implement the testing for all the projects that have a demo available and ramp up the number of demos from there.

Reply:
Just discussed during our planning meeting that we can implement this later.